data_model/src/lib.rs

```diff
@@ -94,13 +94,16 @@ impl Name {
     /// Fails if parsing fails
     #[inline]
     pub fn new(name: &str) -> Result<Self, ParseError> {
-        name.parse::<Self>()
+        Self::from_str(name)
     }
 
     /// Instantly construct [`Name`] assuming `name` is valid.
     #[inline]
     #[allow(clippy::expect_used)]
+    #[cfg(any(test, feature = "cross_crate_testing"))]
     pub fn test(name: &str) -> Self {
+        // I'm not convinced we need these even for testing. Ideally
+        // we should *just* make `Name::new` `const`.
```

Comment:
```suggestion
        // NOTE: I'm not convinced we need these even for testing. Ideally
        // we should *just* make `Name::new` `const`.
```

Reply:
I agree that we don't need this. I thought it was for creating names without the check akin to `new_unchecked`

Reply:
I'd like if you would remove it and all other `test` methods(like `Domain::test` or `Account::test`) entirely if @s8sato agrees

Reply:
I don't mind.
What I intended to provide was short writing without `.expect()` or `.unwrap()`

Reply:
BTW `Name::test` is not unchecked. It panics by wrong input.
I still agree with the removal though